CMakeLists.txt

```diff
@@ -341,13 +341,10 @@ zephyr_cc_option(-Wpointer-arith)
 
 # Declare MPU userspace dependencies before the linker scripts to make
 # sure the order of dependencies are met
-if(CONFIG_CPU_HAS_MPU AND CONFIG_USERSPACE)
-  if(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_APP_SHARED_MEM )
+if(CONFIG_APP_SHARED_MEM AND CONFIG_USERSPACE)
     set(APP_SMEM_DEP app_smem_linker)
-  endif()
-  if(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_ARC AND CONFIG_APPLICATION_MEMORY)
-    set(ALIGN_SIZING_DEP app_sizing_prebuilt linker_app_sizing_script)
-  endif()
+endif()
+if(CONFIG_CPU_HAS_MPU AND CONFIG_USERSPACE)
```

Comment:
Here it should be CONFIG_MEMORY_PROTECTION instead of CONFIG_CPU_HAS_MPU

Reply:
We need the configuration option, not the hw capabilities

Reply:
updated.